plugin-server/src/worker/ingestion/action-matcher.ts

```diff
@@ -137,16 +138,55 @@ export class ActionMatcher {
         this.statsd = statsd
     }
 
-    /** Get all actions matched to the event. */
     public async match(event: PostIngestionEvent, elements?: Element[]): Promise<Action[]> {
+        const legacyResponse = await this.matchLegacy(event, elements)
+        const hogVMResponse = this.matchBytecode(event, elements)
+
+        if (
+            legacyResponse.length !== hogVMResponse.length ||
+            !legacyResponse.every((action, index) => action.id == hogVMResponse[index]?.id)
```

Comment:
Is the order of actions guaranteed to be the same for both matching variants? Just to be safe it could be worth sorting the action IDs first

Reply:
I swapped the matches around:

```ts
const hogVMResponse = this.matchBytecode(event, elements)
const legacyResponse = await this.matchLegacy(event, elements) // this one was first
```

Now both of them call

```ts
const teamActions: Action[] = Object.values(this.actionManager.getTeamActions(event.teamId))
```

in succession before the await has a chance to mess up the order. Thus I'm reasonably sure the order will be the same.